lib/PartitionedProducerImpl.cc

```diff
@@ -443,6 +445,7 @@ void PartitionedProducerImpl::handleGetPartitions(Result result,
                 producers_.push_back(producer);
             }
             // `runPartitionUpdateTask()` will be called in `handleSinglePartitionProducerCreated()`
+            onPartitionsChange(getTopic(), newNumPartitions);
```

Comment:
Unlock `producersLock` before `onPartitionsChange` because it might took long time and it doesn't need to be protected by `producersLock`.